src/custom-element.js

```diff
@@ -591,19 +591,9 @@ function createBaseCustomElementClass(win) {
         }
       }
 
-      if (this.isLoadingEnabled_()) {
-        if (this.isInViewport_) {
-          // Already in viewport - start showing loading.
-          this.toggleLoading(true);
-        } else if (
-          layoutBox.top < PREPARE_LOADING_THRESHOLD &&
-          layoutBox.top >= 0 &&
-          !this.loadingContainer_
-        ) {
-          // Few top elements will also be pre-initialized with a loading
-          // element.
-          this.mutateOrInvoke_(() => this.prepareLoading_());
-        }
+      // Few top elements will also be pre-initialized with a loading element.
+      if (layoutBox.top < PREPARE_LOADING_THRESHOLD && layoutBox.top >= 0) {
```

Comment:
Great observation. 

AFAICT, "toggle loading on measure" exists to avoid the 100ms delay in the `viewportCallback` code path. Looks like immediacy is also the rationale for `PREPARE_LOADING_THRESHOLD`.

Fast forward to today, our new loaders have a delayed UX by design (waits a minimum of 600ms before display -- see `LOADER_APPEAR_TIME`). So I think we can safely remove this remaining code as well.

To make this change truly UX neutral, we can store a timestamp in `viewportCallback` and pass that to `createLoaderElement()` as the starting time. Less code FTW!

/cc @dvoytenko

Reply:
Thank you for the historical context! That there has been an evolution in loader UX explains the code well.

> exists to avoid the 100ms delay in the viewportCallback code path

Now that delaying the loader is part of the loader service itself, can we remove this logic as well?

> To make this change truly UX neutral, we can store a timestamp in viewportCallback and pass that to createLoaderElement() as the starting time. Less code FTW!

`createLoaderElement` already records the current timestamp when called (to measure the delay in installing the `amp-loader` service). I think removing the 100ms delay would mean that we don't need to do any extra recording?

Reply:
> can we remove this logic as well?

Yup! I suggested this above.

> I think removing the 100ms delay would mean that we don't need to do any extra recording?

Correct, but we probably want to preserve the behavior where we skip elements that enter/exit viewport within 100ms (very fast scrolling).

Reply:
Can we remove this (and PREPARE_LOADING_THRESHOLD)? We may just need a single call to fetch the amp-loader extension, but that can be done in the viewportCallback codepath.